squirrel.frontier/src/main/java/org/dice_research/squirrel/frontier/recrawling/FrontierQueryGenerator.java

```diff
@@ -0,0 +1,52 @@
+package org.dice_research.squirrel.frontier.recrawling;
+
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FrontierQueryGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FrontierQueryGenerator.class);
+
+    private FrontierQueryGenerator() {
+    }
+
+    /**
+     * Return outdated uris by comparing their endtime stamps.
+     * @return All triples with time stamp in the default graph.
+     */
+
+
+
+    public static Query getOutdatedUrisQuery() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("PREFIX  sq:   <http://w3id.org/squirrel/vocab#>\n" +
+            "PREFIX  prov: <http://www.w3.org/ns/prov#>\n" +
+            "PREFIX  xsd:  <http://www.w3.org/2001/XMLSchema#>"
+            + "SELECT ?uri  WHERE { \n "+
+            "{\n" +
+            "SELECT ?uri ?endtime (NOW() - (?endtime) AS ?diff)\n" +
+            "WHERE{\n" +
+            "\n" +
+            "  {\n" +
+            "    SELECT  ?uri  (MAX(?timestamp) as ?endtime)\n" +
+            "    WHERE\n" +
+            "    { \n" +
+            "        ?s  sq:crawled  ?uri ;\n" +
+            "        prov:endedAtTime  ?timestamp.\n" +
+            "\n" +
+            "    }\n" +
+            "    GROUP BY ?uri\n" +
+            "  } \n" +
+            "}\n" +
+            "}\n" +
+            "FILTER(?diff > \"60\"^^xsd:double)}\n" +
+            "");
```

Comment:
Using a StringBuilder does not really make sense if you are using the `+` operator to create a single, large String that is added to the builder. The builder will not really have anything to do...
In general, it is ok to build a String using only the `+` operator as long as all parts of the string are constants. The compiler will concatenate all of them already when compiling.